test/test_torch.py

```diff
@@ -12072,6 +12072,48 @@ def test_memory_format_permute_cuda(self):
         y = nhwc.permute(0, 1, 3, 2).permute(0, 1, 3, 2)
         self.assertFalse(y.is_contiguous(memory_format=torch.channels_last))
 
+    def _test_memory_format_empty_like(self, x):
+        nhwc = x.contiguous(memory_format=torch.channels_last)
+
+        like = torch.empty_like(nhwc, memory_format=torch.preserve_format)
```

Comment:
I feel that these tests are not sufficient. Suppose I write a buggy implementation of the stride computation so that I put channels before batch rather than after.  I don't think any of these tests would catch an error like that.

Reply:
There is some small utility in replicating the stride logic for a specific dimensionality here, because when you generalize the formula it will help catch errors in a less correlated way. Other properties you could check are that transpositions related the striding between channels first and channels last.

Reply:
Agree, but it will make sense as part of the next PR where I plan to introduce this channels_last criteria, and cover basic operations on it.